index.ts

```diff
@@ -45,5 +45,26 @@ export function patternToRegex(...matchPatterns: readonly string[]): RegExp {
 		return allStarsRegex;
 	}
 
-	return new RegExp(matchPatterns.map(x => getRawRegex(x)).join('|'));
+	return new RegExp(matchPatterns.map(x => getRawPatternRegex(x)).join('|'));
+}
+
+function getRawGlobRegex(glob: string): string {
+	let regexString = glob;
+	regexString = regexString.startsWith('*') ? regexString.replace(/^[*]+/, '') : '^' + regexString;
+	regexString = regexString.endsWith('*') ? regexString.replace(/[*]+$/, '') : regexString + '$';
+
+	return regexString
+		.replace(/[/]/g, '[/]') // Escape slashes
+		.replace(/[.]/g, '[.]') // Escape dots
+		.replace(/[*]+/g, '.*') // Wildcard
+		.replace(/[?]/g, '.'); // Single character
```

Comment:
This is all that's needed, really. The previous 2 lines are just to avoid starting with `^.*`, which is equivalent to not existing

Reply:
I have to look at it more clearly later but I'm pretty sure the ? Matches one or zero

Reply:
[MDN:](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/content_scripts#globs)

> ? matches exactly one character.


[Chrome:](https://developer.chrome.com/docs/extensions/mv3/content_scripts/#matchAndGlob)

> the question mark ? matches any single character.

I was able to confirm with the demo extension in Chrome that `?` does not match missing characters

Reply:
However I'll likely have to escape more characters that may appear in search parameters, like `[`

Reply:
Yeah and I did it on chrome last time

Reply:
> However I'll likely have to escape more characters that may appear in search parameters, like `[`

Why not escape everything like I did which is much safer in my opinion